Replace AuthToken with routed job attrs (SOFTWARE-5185)

edquist opened the pull request:
Per SOFTWARE-5185, the AuthToken* attributes we used for VO info in the
htcondor-ce probe don't exist in the routed job for non-condor batch
systems.  And, even worse, these may have entirely different values for
condor batch systems (a different token can be used to authN to the local
schedd).

In HTCONDOR-1071, htcondor-ce config is added to copy the pre-routed
job's AuthToken* attrs to orig_AuthToken* attrs in the routed job.

Thus, SOFTWARE-5185 calls for replacing the AuthToken* references in the
htcondor-ce probe with orig_AuthToken*.  In order to make this
probe-specific, I add a probe_name check, as the condor-ap and
htcondor-ce probes share the same probe code.

Note also that AuthToken* references exist in common/condor_ce.py,
in certinfo code (which seems to be unimportant at this point, since
it's going away), but also in CheckAndExtendUserIdentity from
common/xml_utils.py, to fill in missing vo info for jobs by querying the
htcondor-ce directly.  Do these need to get the "orig_" prefix also?
(Always? Just when running the htcondor-ce probe?  Never?)

...

It _seems_ to me that for the condor_ce.py code we just need to update the attr names in the condor_ce_q query, which is made against routed jobs - and that being the case i always query the `orig_`-prefixed names (not just in the htcondor-ce probe case).

edquist committed:
replace AuthToken with routed job attrs (SOFTWARE-5185)

Per SOFTWARE-5185, the AuthToken* attributes we used for VO info in the
htcondor-ce probe don't exist in the routed job for non-condor batch
systems.  And, even worse, these may have entirely different values for
condor batch systems (a different token can be used to authN to the local
schedd).

In HTCONDOR-1071, htcondor-ce config is added to copy the pre-routed
job's AuthToken* attrs to orig_AuthToken* attrs in the routed job.

Thus, SOFTWARE-5185 calls for replacing the AuthToken* references in the
htcondor-ce probe with orig_AuthToken*.  In order to make this
probe-specific, I add a probe_name check, as the condor-ap and
htcondor-ce probes share the same probe code.

Note also that AuthToken* references exist in common/condor_ce.py,
in certinfo code (which seems to be unimportant at this point, since
it's going away), but also in CheckAndExtendUserIdentity from
common/xml_utils.py, to fill in missing vo info for jobs by querying the
htcondor-ce directly.  Do these need to get the "orig_" prefix also?
(Always? Just when running the htcondor-ce probe?  Never?)

edquist committed:
retrieve orig_AuthToken* attrs in condor_ce_q query

as these only apply to the routed jobs, which have the AuthToken*
attrs copied with the orig_ prefix.

they are left as-is (AuthToken*) in the formatting, so that they
can be handled by the internal job info representation without
modification to the rest of the plumbing

brianhlin reviewed (approved):
> Thus, SOFTWARE-5185 calls for replacing the AuthToken* references in the
htcondor-ce probe with orig_AuthToken*. In order to make this
probe-specific, I add a probe_name check, as the condor-ap and
htcondor-ce probes share the same probe code.

This is fine, though I don't think that the `AuthToken*` attrs will have any semantic meaning for the `condor-ap` probe, at least not yet.

> Note also that AuthToken* references exist in common/condor_ce.py,
in certinfo code (which seems to be unimportant at this point, since
it's going away), but also in CheckAndExtendUserIdentity from
common/xml_utils.py, to fill in missing vo info for jobs by querying the
htcondor-ce directly. Do these need to get the "orig_" prefix also?
(Always? Just when running the htcondor-ce probe? Never?)

I say we change all of the references, just in case. I don't recall what `CheckAndExtendUserIdentity` hooks into but we certainly only added the `AuthToken*` attrs as a replacement for the `x509userproxy*` attrs for extracting VO data.

Otherwise, LGTM, pre-approving

edquist commented:
> I don't think that the `AuthToken*` attrs will have any semantic meaning for the `condor-ap` probe, at least not yet.

If that's the case, i can remove the conditionalization and always use the `orig_AuthToken*` names, regardless of the `probe_name` (that is, keep it for condor-ap).

> I say we change all of the references, just in case

Okie doke.

edquist committed:
use orig_AuthToken* names everywhere (SOFTWARE-5185)

for good measure, per blin

edquist committed:
do not conditionalize orig_/AuthToken* on probe_name (SOFTWARE-5185)

because it seems the AuthToken* attrs do not have any semantic meaning
in the condor-ap probe, at the moment
